crates/wascc-provider/src/lib.rs

```diff
@@ -422,12 +432,73 @@ fn wascc_run(
     );
     capabilities.push(Capability {
         name: LOG_CAPABILITY,
+        binding: None,
         env: logenv,
     });
 
     let load = Actor::from_bytes(data).map_err(|e| anyhow::anyhow!("Error loading WASM: {}", e))?;
     let pk = load.public_key();
 
+    if load.capabilities().contains(&FS_CAPABILITY.to_owned()) {
+        for vol in &volumes {
+            info!(
+                "Loading File System capability for volume name: '{}' host_path: '{}'",
+                vol.name,
+                vol.host_path.display()
+            );
+            let mut fsenv: HashMap<String, String> = HashMap::new();
+            fsenv.insert(
+                "ROOT".to_owned(),
+                vol.host_path.as_path().to_str().unwrap().to_owned(),
+            );
+            let fs_provider = FileSystemProvider::new();
+            let fs_capability =
+                NativeCapability::from_instance(fs_provider, Some(vol.name.clone())).map_err(
+                    |e| anyhow::anyhow!("Failed to instantiate File System capability: {}", e),
+                )?;
+            host.lock()
+                .unwrap()
+                .add_native_capability(fs_capability)
+                .map_err(|e| anyhow::anyhow!("Failed to add File System capability: {}", e))?;
+            capabilities.push(Capability {
+                name: FS_CAPABILITY,
+                binding: Some(vol.name.clone()),
+                env: fsenv,
+            });
+        }
+    }
+
+    // wascc has native and portable capabilities.
+    //
+    // Native capabilities are either dynamic libraries (.so, .dylib, .dll)
+    // or statically linked Rust libaries. If the native capabilty is a dynamic
+    // library it must be loaded and configured through [`NativeCapability::from_file`].
+    // If it is a statically linked libary it can be configured through
+    // [`NativeCapability::from_instance`].
+    //
+    // Portable capabilities are WASM modules.  Portable capabilities
+    // don't fully work, and won't until the WASI spec has matured.
+    //
+    // Here we are using the native capabilties as statically linked libraries that will
+    // be compiled into the wascc-provider binary.
+    info!("Loading HTTP capability");
+    let http_provider = HttpServerProvider::new();
+    let data = NativeCapability::from_instance(http_provider, None)
```

Comment:
My understanding was that only one http provider should be run at a time. Did that only apply to the shared object files?

Reply:
Good point. I was trying some things out earlier. We can put this back where it was now